### ISSUE TITLE
ngx-translate with nrwl

### ISSUE THREAD

**Issue by BVSureshBabu** [ngx-translate/core#1082]
Hi,

I am using ngx-translate with nrwl. It is working fine if I keep all translation file in app and loading all once. **But I want to load translation files per lib lazily i.e whenever module loads that time I want to load translation files loaded in that lib**. libs are lazyloaded. I am using below versions

    "@nrwl/nx": "1.0.0",
    "@ngx-translate/core": "9.1.1",
    "@ngx-translate/http-loader": "2.0.1",
    "@angular/compiler-cli": "5.2.7",

Please help me to achieve this. Thanks in advance.

Thanks & Regards,
Suresh

**Comment by chrismclarke** [comment]
I'm trying to achieve the same, I suggest you open a question on stackoverflow to discuss there as I don't think this is an issue with the ngx-translate platform

**Comment by Salahudeen** [comment]
> Hi,
> 
> I am using ngx-translate with nrwl. It is working fine if I keep all translation file in app and loading all once. **But I want to load translation files per lib lazily i.e whenever module loads that time I want to load translation files loaded in that lib**. libs are lazyloaded. I am using below versions
> 
> ```
> "@nrwl/nx": "1.0.0",
> "@ngx-translate/core": "9.1.1",
> "@ngx-translate/http-loader": "2.0.1",
> "@angular/compiler-cli": "5.2.7",
> ```
> 
> Please help me to achieve this. Thanks in advance.
> 
> Thanks & Regards,
> Suresh

Did you find a solution for the same? I am also looking for a way to achieve the same...can you help?

**Comment by docwhite** [comment]
Same. I am using `nx`. Here's the further I have got. Still can't have the behaviour to really `extend` but I think I got pretty close, evolving from other people's StackBlitz. Have a play, maybe you see something else we can do: https://stackblitz.com/edit/translations-and-lazy-loading?file=README.md

**Comment by vmagalhaes** [comment]
@docwhite any updates of this?

**Comment by docwhite** [comment]
Sadly not @vmagalhaes. I ended up learning another tool called transloco.

**Comment by vmagalhaes** [comment]
> Sadly not @vmagalhaes. I ended up learning another tool called transloco.

And did you had a better result with lib <> app interaction?

**Comment by docwhite** [comment]
@vmagalhaes yes you can craft it using what they call "scopes" https://ngneat.github.io/transloco/docs/lazy-load/scope-configuration/

**Comment by rbalet** [comment]
Fixed in the latest [v17 release](https://github.com/ngx-translate/core/releases/tag/v17.0.0) with the new isolated & extended logic

-> https://ngx-translate.org/reference/concepts/#configuration
